### PR TITLE
Merge in https://github.com/aryan26roy/fastjet

### PR DESCRIPTION
**Pull request by jpivarski**
The first few months of this project were developed in @aryan26roy's personal repo, but now that the build procedure works, I'm moving everything here without history. (Some mistakes were made in the early history; it's best to start with a clean git.)

At the end of this PR, it should be ready to use for all further development.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+# Warning: do not change the path here. To use autodoc, you need to install the
+# package first.
+
+from typing import List
+
+# -- Project information -----------------------------------------------------
+
+project = "fastjet"
+copyright = "2021, Scikit-HEP"
```
Comment (eduardo-rodrigues) [inline docs/conf.py:15]:
```suggestion
copyright = "2021, The Scikit-HEP admins"
```

Comment (jpivarski) [reply docs/conf.py:15]: https://github.com/scikit-hep/cookie is generating this by default; if you prefer the shorter version, also update it there.

**Comment by eduardo-rodrigues** [comment]
Hi @jpivarski, @aryan26roy, very nice to see this coming in 👍 .

File: docs/conf.py (hunk 1 above)
Comment (eduardo-rodrigues) [reply docs/conf.py]: Thanks, have just made the suggestion to Henry.

**Comment by henryiii** [comment]
Check-manifest requires a recursive checkout.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - master
+    - main
+    - develop
+
+jobs:
+  pre-commit:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
```
Comment (henryiii) [inline .github/workflows/ci.yml:17]:
```suggestion
    - uses: actions/checkout@v2
       with:
         submodules: recursive
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - uses: actions/setup-python@v2
+    - uses: pre-commit/action@v2.0.0
```
Comment (henryiii) [inline .github/workflows/ci.yml:19]:
```suggestion
    - uses: pre-commit/action@v2.0.2
```

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  hooks:
+  - id: flake8
+    exclude: docs/conf.py
+    additional_dependencies: [flake8-bugbear, flake8-print]
+
+# - repo: https://github.com/mgedmin/check-manifest
+#   rev: "0.45"
+#   hooks:
+#   - id: check-manifest
+#     stages: [manual]
```
Comment (henryiii) [inline .pre-commit-config.yaml:48]:
```suggestion
- repo: https://github.com/mgedmin/check-manifest
  rev: "0.45"
  hooks:
  - id: check-manifest
    stages: [manual]
```

File: .github/dependabot.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+version: 2
+updates:
+  # Maintain dependencies for GitHub Actions
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    ignore:
+      # Offical actions have moving tags like v1
```
Comment (henryiii) [inline .github/dependabot.yml:9]:
```suggestion
      # Official actions have moving tags like v1
```

Pretty sure this is my typo :)

Comment (jpivarski) [reply .github/dependabot.yml:9]: These things are coming directly from scikit-hep/cookie, which is why I had overlooked recursive checkouts.

I'll apply all of these as a single commit on my side. Thanks!

**Comment by henryiii** [comment]
You need submodules v2 to use the `submodules: recursive` syntax. `v1` called it something else. For `v2` you need `fetch-depth: 0` if you use setuptools_scm.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (jpivarski) [reply .github/workflows/ci.yml]: Actually, Fastjet has a submodule within it, for SISCone, so I don't want to limit the recursive depth.

**Comment by jpivarski** [comment]
> You need submodules v2 to use the `submodules: recursive` syntax. `v1` called it something else. For `v2` you need `fetch-depth: 0` if you use setuptools_scm.

If `setuptools_scm` is what sets version numbers, I've disabled it for this repo because we want the version numbers to correlate with FastJet version numbers. For instance, the first version of this package will be 3.3.4.0rc1, where the fourth digit is the Python interface version.

**Comment by henryiii** [comment]
Yes, then you don't need `fetch-depth: 0` (it'll be slightly faster without it).

File: .github/workflows/ci.yml (hunk 2 above)
Comment (henryiii) [reply .github/workflows/ci.yml]: There are only two options. `submodules: true` checkout out one level, and `submodules: recursive` checks out all levels. `fetch-depth` refers to the git history, not submodules.

**Comment by henryiii** [comment]
> Rename fastjet -> fastjet-core to prevent unintended import in Python.

One of the reasons I always use an `/extern` directory. :)

**Comment by jpivarski** [comment]
I don't know what's happening with

```
2021-04-16T00:51:31.4192598Z   checking CGAL in /tmp/pip-req-build-5mdakdlc/CGAL-5.2.1... lib
2021-04-16T00:51:31.6587274Z   checking CGAL/Exact_predicates_inexact_constructions_kernel.h presence... no
```

when the same task has

```
2021-04-16T00:51:22.5850739Z   -rw-r--r-- 1 runner docker 1586 Apr 16 00:51 /tmp/pip-req-build-5mdakdlc/CGAL-5.2.1/include/CGAL/Exact_predicates_inexact_constructions_kernel.h
```

and exactly the same thing always succeeds ("`presence... yes`") on my computer. I'm going to merge this just in case it's something to do with the submodule version on main vs the submodule version on this branch (i.e. if it's getting the wrong fastjet-core).

**Comment by jpivarski** [comment]
Oh, I couldn't merge because it was a draft. Well, wait for success anyway.

**Pull request merged**